src/api/c++/z3++.h

```diff
@@ -105,11 +110,23 @@ namespace z3 {
     */
     class config {
         Z3_config    m_cfg;
-        config(config const & s);
```

Comment:
These are just declarations, not definitions. This could lead to linker errors and other issues. Added `=default` so that the compiler will include a definition if it was silently implicitly deleted by something else (for example due to a rule of 5 violation)

Reply:
these changes are wrong. if anything, they would be marked with = delete. That's the way of doing things

Reply:
In the original source they were:
```
        config(config const & s);
        config & operator=(config const & s);
```
Are you saying that's a bug?

Reply:
Looking over the code it does seem that these should be deleted to avoid the reference counting error that would occur via double deletion. I will change them to `= delete;`, both the copy constructor and assignment operator.

Reply:
Oh, I see, they were declared private. Got it. I will rectify this in all related comments.